vividus/src/main/java/org/vividus/expression/RoundExpressionProcessor.java

```diff
@@ -18,90 +18,59 @@
-                + "(?:,\\s*(%s))?\\))$", roundingModes);
-        ROUND_EXPRESSION_PATTERN = Pattern.compile(pattern, Pattern.CASE_INSENSITIVE);
-    }
+    private static final Pattern VALUE_TO_ROUND_PATTERN = Pattern.compile(
+            "[-+]?(\\d+\\.?\\d*|(\\.\\d+))(?:[Ee]+[-+]?\\d+)?");
+    private static final Pattern MAX_FRACTION_DIGITS_PATTERN = Pattern.compile("\\d+");
+    private static final Pattern ROUNDING_MODE_PATTERN = Stream.of(RoundingMode.values())
+            .map(RoundingMode::name)
+            .map(String::toLowerCase)
+            .map(e -> e.replaceAll("_", "[ _]"))
+            .collect(Collectors.collectingAndThen(Collectors.joining("|"), Pattern::compile));
 
     public RoundExpressionProcessor(FluentEnumConverter fluentEnumConverter)
     {
-        super(ROUND_EXPRESSION_PATTERN);
-        this.fluentEnumConverter = fluentEnumConverter;
-    }
-
-    @Override
-    protected String evaluateExpression(Matcher expressionMatcher)
-    {
-        RoundExpression roundExpression = new RoundExpression(expressionMatcher, fluentEnumConverter);
-        return new BigDecimal(roundExpression.getValue())
-                .setScale(roundExpression.getMaxFractionDigits(), roundExpression.getRoundingMode())
-                .stripTrailingZeros()
-                .toPlainString();
-    }
-
-    private static final class RoundExpression
-    {
-        private static final int VALUE_GROUP = 1;
-        private static final int MAX_FRACTION_DIGITS_GROUP = 2;
-        private static final int ROUNDING_MODE_GROUP = 3;
-
-        private static final int DEFAULT_MAX_FRACTION_DIGITS = 2;
-
-        private final FluentEnumConverter converter;
-        private final String value;
-        private final String maxFractionDigits;
-        private final String roundingMode;
-
-        RoundExpression(Matcher durationMatcher, FluentEnumConverter fluentEnumConverter)
-        {
-            converter = fluentEnumConverter;
-            value = durationMatcher.group(VALUE_GROUP);
-            maxFractionDigits = durationMatcher.group(MAX_FRACTION_DIGITS_GROUP);
-            roundingMode = durationMatcher.group(ROUNDING_MODE_GROUP);
-        }
-
-        public String getValue()
-        {
-            return value;
-        }
-
-        public int getMaxFractionDigits()
-        {
-            return NumberUtils.isCreatable(maxFractionDigits) ? Integer.parseInt(maxFractionDigits)
-                    : DEFAULT_MAX_FRACTION_DIGITS;
-        }
-
-        public RoundingMode getRoundingMode()
-        {
-            if (roundingMode == null)
-            {
-                return value.charAt(0) == '-' ? RoundingMode.HALF_DOWN : RoundingMode.HALF_UP;
-            }
-            return (RoundingMode) converter.convertValue(roundingMode, RoundingMode.class);
-        }
+        super("round", 1, MAX_ARGS, args -> {
+            String valueToRound = args.get(0);
```

Comment:
in general, does it make sense to keep indexes under static variable names? meaning that previously arg 0 was VALUE, arg 1 MAX_FRACTION_DIGITS etc., I understand that variables name that we assign are alear enough, but just to be safe perhaps we could continue name them

Reply:
discussed